### DISCUSSION TITLE
[autobahn] IConnectionOptions is missing error handler callbacks

### DISCUSSION THREAD

**Discussion by trombipeti** [DefinitelyTyped/DefinitelyTyped#55688] [Issues with a @types package] [answered]
The IConnectionOptions interface is missing the error handler callbacks. From what I could see, their signature is:
`on_user_error(error: object | Error, customErrorMessage: string | undefined) => void`
`on_internal_error(error: object | Error, customErrorMessage: string | undefined) => void`

These callbacks exist since 2017. Is there any specific reason why they're not in the interface, or did just nobody need them yet? In the second case I'd be more than happy to do a PR.

https://github.com/crossbario/autobahn-js/blob/bd250753432a7bc61b7b7dae1df5fc70ddedcad6/packages/autobahn/lib/util.js#L354
https://github.com/crossbario/autobahn-js/blob/bd250753432a7bc61b7b7dae1df5fc70ddedcad6/packages/autobahn/lib/connection.js#L271

[a comment by an automated account is not shown]

**Comment by darkl** [comment, marked as answer]
Well, I wrote these type definitions in 2015, so that member didn't exist
at that time.

Feel free to submit a pull request.

Elad

**Reply by trombipeti** [reply]
Oh well, I'll do that then (maybe also check for other changes in the last 6 years or so ^^).

Thanks!
